sklbench/benchmarks/sklearn_estimator.py

```diff
@@ -0,0 +1,471 @@
+    accuracy_score,
+    balanced_accuracy_score,
+    completeness_score,
+    davies_bouldin_score,
+    homogeneity_score,
+    log_loss,
+    mean_squared_error,
+    r2_score,
+    roc_auc_score,
+)
+
+from ..datasets import load_data
+from ..datasets.transformer import split_and_transform_data
+from ..utils.bench_case import get_bench_case_value, get_data_name
+from ..utils.common import custom_format, get_module_members
+from ..utils.config import bench_case_filter
+from ..utils.custom_types import BenchCase, Numeric, NumpyNumeric
+from ..utils.logger import logger
+from ..utils.measurement import measure_case
+from ..utils.special_params import assign_case_special_values_on_run
+from .common import main_template
+
+
+def get_estimator(library_name: str, estimator_name: str):
+    classes_map, _ = get_module_members(library_name.split("."))
+    if estimator_name not in classes_map:
+        raise ValueError(
+            f"Unable to find {estimator_name} estimator in {library_name} module."
+        )
+    if len(classes_map[estimator_name]) != 1:
+        logger.debug(
+            f'List of estimator with name "{estimator_name}": '
+            f"{classes_map[estimator_name]}"
+        )
+        logger.warning(
+            f"Found {len(classes_map[estimator_name])} classes for "
+            f'"{estimator_name}" estimator name. '
+            f"Using first {classes_map[estimator_name][0]}."
+        )
+    estimator = classes_map[estimator_name][0]
+    if not issubclass(estimator, BaseEstimator):
+        logger.info(f"{estimator} estimator is not derived from sklearn's BaseEstimator")
+    return estimator
+
+
+def get_estimator_methods(bench_case: BenchCase) -> Dict[str, List[str]]:
+    # default estimator methods
+    estimator_methods = {
+        "training": ["fit"],
+        "inference": ["predict", "predict_proba", "transform"],
+    }
+    for stage in estimator_methods.keys():
+        methods = get_bench_case_value(
+            bench_case, f"algorithm:estimator_methods:{stage}", None
+        )
+        if methods is not None:
+            estimator_methods[stage] = methods.split("|")
+    return estimator_methods
+
+
+def estimator_to_task(estimator_name: str) -> str:
+    estimator_groups = {
+        "clustering": ["DBSCAN", "KMeans"],
+        "decomposition": ["PCA"],
+        "classification": ["LogisticRegression", "SVC", "NuSVC"],
+        "regression": [
+            "LinearRegression",
+            "Ridge",
+            "Lasso",
+            "ElasticNet",
+            "SVR",
+            "NuSVR",
+        ],
+        "manifold": ["TSNE"],
+    }
+    if estimator_name.endswith("Classifier"):
+        return "classification"
+    elif estimator_name.endswith("Regressor"):
+        return "regression"
+    else:
+        for group_name, estimators_in_group in estimator_groups.items():
+            if estimator_name in estimators_in_group:
+                return group_name
+    return "unknown"

```

Comment:
I think it is bad design.
This kind of meta data should be provided in config files, or somewhere else.
If we are going to extend our benchmarks, the only files that we should update: datasets and config files.
It is not obvious that we should changes some specific internal functions in the tool.
For example, if we are going to extend benchmark tools for EmpiricalCovariance, what we should do in this case.

Reply:
Mapping was moved to separate `estimator_task_map.json` file near benchmark.